File: WaterManager/src/main/java/pissir/watermanager/model/item/Misura.java
package pissir.watermanager.model.item;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author Almasio Luca
 * @author Borova Dritan
 * @author Gattico Alessandro
 */
@Setter
@Getter
@NoArgsConstructor
public class Misura {
	
	private int id;
	private Double value;
	private String time;
	private int idSensore;
	
	
	public Misura (int id, Double value, String time, int idSensore) {
		this.id = id;
		this.value = value;
		this.time = time;
		this.idSensore = idSensore;
	}
	
	
	public Misura(Double value, String time, int idSensore) {
		this.value = value;
		this.time = time;
		this.idSensore = idSensore;
	}
	
}

File: WaterManagerUI/Model/Item/Misura.cs
using Newtonsoft.Json;

namespace WaterManagerUI.Model.Item;

public class Misura
{
    public int id { get; set; }
    public Double value { get; set; }
    public String time { get; set; }
    public int idSensore { get; set; }


    public Misura(Double value, String time, int idSensore)
    {
        this.id = 0;
        this.value = value;
        this.time = time;
        this.idSensore = idSensore;
    }

    [JsonConstructor]
    public Misura(int id, double value, string time, int idSensore)
    {
        this.id = id;
        this.value = value;
        this.time = time;
        this.idSensore = idSensore;
    }
}